src/panels/KaitoModelsPanel.ts

```diff
@@ -0,0 +1,257 @@
+            cancelRequest: () => {
+                this.cancel();
+            },
+        };
+    }
+
+    private async handleGenerateCRDRequest(yaml: string) {
+        const doc = await vscode.workspace.openTextDocument({
+            content: yaml,
+            language: "yaml",
+        });
+        vscode.window.showTextDocument(doc);
+    }
+
+    nullIsFalse(value: boolean | null): boolean {
+        return value ?? false;
+    }
+
+    private async handleDeployKaitoRequest(model: string, yaml: string, webview: MessageSink<ToWebViewMsgDef>) {
+        this.cancelToken = false;
+        try {
+            const tempFilePath = join(tmpdir(), `kaito-deployment-${Date.now()}.yaml`);
+            writeFileSync(tempFilePath, yaml, "utf8");
+            const command = `apply -f ${tempFilePath}`;
+            const kubectlresult = await invokeKubectlCommand(this.kubectl, this.kubeConfigFilePath, command);
+            if (failed(kubectlresult)) {
+                vscode.window.showErrorMessage(
+                    `There was an error with deployment. Try deploying again. ${kubectlresult.error}`,
+                );
+                return;
+            }
+        } catch (error) {
+            vscode.window.showErrorMessage(`Error during deployment: ${error}`);
+        }
+        this.handleUpdateStateRequest(model, webview);
+        let progress = await this.getProgress(model);
+        while (!this.nullIsFalse(progress.workspaceReady) && !this.cancelToken) {
+            await this.handleUpdateStateRequest(model, webview);
+            progress = await this.getProgress(model);
+        }
+        if (this.cancelToken) {
+            await this.handleResetStateRequest(webview);
+        } else {
+            await this.handleUpdateStateRequest(model, webview);
+        }
+    }
+
+    private async handleWorkspaceExistsRequest(model: string) {
+        const command = `get workspace ${model} -w`;
+        const kubectlresult = await invokeKubectlCommand(this.kubectl, this.kubeConfigFilePath, command);
+        if (failed(kubectlresult)) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+    private async handleResetStateRequest(webview: MessageSink<ToWebViewMsgDef>) {
+        webview.postDeploymentProgressUpdate({
+            clusterName: this.clusterName,
+            modelName: "",
+            workspaceExists: false,
+            resourceReady: null,
+            inferenceReady: null,
+            workspaceReady: null,
+            age: 0,
+        });
+    }
+    convertAgeToMinutes(creationTimestamp: string): number {
+        const createdTime = new Date(creationTimestamp);
+        const currentTime = new Date();
+        const differenceInMilliseconds = currentTime.getTime() - createdTime.getTime();
+        const differenceInMinutes = Math.floor(differenceInMilliseconds / 1000 / 60);
+        return differenceInMinutes;
+    }
+    statusToBoolean(status: string): boolean {
```

Comment:
💡 https://github.com/Azure/vscode-aks-tools/pull/963#pullrequestreview-2365020283

also some recommendations here: https://github.com/Azure/vscode-aks-tools/pull/963#discussion_r1792833650

Reply:
📓 Nice, seems like @tejhan mentioned that he will take care of most of this in PR here post this merge: https://github.com/Azure/vscode-aks-tools/pull/1003 , leaving this as fyi comment.